src/App.tsx

```diff
@@ -1,17 +1,100 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
-import users from './api/users';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { FullTodo, Todo, User } from './react-app-env';
+import TodoList from './components/TodoList/TodoList';
 
 const App: React.FC = () => {
+  const [newUser, setNewUser] = useState('');
+  const [newTitle, setNewTitle] = useState('');
+  const [newCompleted, setNewCompleted] = useState(false);
+  const [newTodo, setNewTodo] = useState(todosFromServer);
+
+  const preparedTodos = (
+    todos: Todo[],
+    users: User[],
+  ): FullTodo[] => {
+    return todos.map(todo => ({
+      ...todo,
+      user: users.find(user => user.id === todo.userId) || null,
+    }));
+  };
+
+  const todos = preparedTodos(
+    newTodo,
+    usersFromServer,
+  );
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <form onSubmit={(event) => {
+        event.preventDefault();
+
+        setNewTodo([...newTodo, {
+          userId: +newUser,
+          id: newTodo.length + 1,
+          title: newTitle,
+          completed: newCompleted,
+        }]);
+
+        setNewTitle('');
+        setNewCompleted(false);
+        setNewUser('');
+      }}
+      >
+        <label
+          className="level level-left m-2"
+          htmlFor="titleInput"
+        >
+          Title of Todo:
+        </label>
+        <input
+          type="text"
+          id="titleInput"
+          className="input is-link level level-left column is-2 ml-1"
+          required
```

Comment:
The task requires that after pressing Add we see `Please enter the title` or `Please choose a user`.  Attribute "required" is incorrect in this case.